### ISSUE TITLE
sampled softmax vs regular softmax

### ISSUE THREAD

**Issue by zcyang** [tensorflow/tensorflow#1231]
For the seq2seq example, sampled softmax is 2-3 times slower than regular softmax, is it because sampled softmax is on CPU while regular softmax is on GPU? Then why do we need sampled softmax, it saves gpu memory?

But the convergence speed of sampled softmax (in terms of iterations) is much faster then regular softmax? Any mathematical reasons?

Many thanks!

**Comment by martinwicke** [comment]
This is a question best suited for stackoverflow. Please ask it there to get a good answer. Regular softmax isn't a good idea for very large vocabularies.

**Comment by zcyang** [comment]
bring this up since I find the loss calculated with sampled softmax is significant lower than regular softmax, this explains why the perplexity is low while the model is not reasonable here #550
